drizzle-schema/src/effect-schema.ts

```diff
@@ -0,0 +1,442 @@
+    columnEntries.map(([name, column]) => {
+      return [name, mapColumnToSchema(column)];
+    }),
+  );
+
+  if (refine) {
+    schemaEntries = Object.assign(
+      schemaEntries,
+      Object.fromEntries(
+        Object.entries(refine).map(([name, refineColumn]) => {
+          return [
+            name,
+            typeof refineColumn === 'function' &&
+              // guard against classes which are also functions, e.g. `Schema.Literal` class
+              !('annotations' in refineColumn)
+              ? refineColumn(schemaEntries as any)
+              : refineColumn,
+          ];
+        }),
+      ),
+    );
+  }
+
+  for (const [name, column] of columnEntries) {
+    if (!column.notNull) {
+      schemaEntries[name] = S.optional(S.NullOr(schemaEntries[name]!)) as any;
+    } else if (column.hasDefault) {
+      schemaEntries[name] = S.optional(schemaEntries[name]!) as any;
+    }
+  }
+
+  return S.Struct(schemaEntries) as any;
+}
+
+// export function createSelectSchema<TTable extends Table>(
+//   table: TTable,
+// ): S.Schema<
+//   Simplify<S.Struct.Constructor<SelectColumnPropertySignatures<TTable>>>
+// > {
+//   const columns = getTableColumns(table);
+//   const columnEntries = Object.entries(columns);
+//
+//   const schemaEntries = Object.fromEntries(
+//     columnEntries.map(([name, column]) => {
+//       return [name, mapColumnToSchema(column)];
+//     }),
+//   );
+//
+//   for (const [name, column] of columnEntries) {
+//     if (!column.notNull) {
+//       schemaEntries[name] = S.NullOr(schemaEntries[name]!);
+//     }
+//   }
+//
+//   return S.Struct(schemaEntries) as any;
+// }
+
+function mapColumnToSchema(column: Column): S.Schema<any, any> {
+  let type: S.Schema<any, any> | undefined;
+
+  if (isWithEnum(column)) {
+    type = column.enumValues.length
+      ? S.Literal(...column.enumValues)
+      : S.String;
+  }
+
+  if (!type) {
+    if (is(column, pgCore.PgUUID)) {
+      type = UUID;
+    } else if (column.dataType === 'custom') {
+      type = S.Any;
+    } else if (column.dataType === 'json') {
+      type = jsonSchema;
+    } else if (column.dataType === 'array') {
+      type = S.Array(
+        mapColumnToSchema((column as PgArray<any, any>).baseColumn),
+      );
+    } else if (column.dataType === 'number') {
+      type = S.Number;
+    } else if (column.dataType === 'bigint') {
+      type = S.BigIntFromSelf;
+    } else if (column.dataType === 'boolean') {
+      type = S.Boolean;
+    } else if (column.dataType === 'date') {
+      type = S.DateFromSelf;
+    } else if (column.dataType === 'string') {
+      let sType = column.notNull ? S.NonEmpty : S.String;
+
+      if (
+        (is(column, pgCore.PgChar) ||
+          is(column, pgCore.PgVarchar) ||
+          is(column, mysqlCore.MySqlVarChar) ||
+          is(column, mysqlCore.MySqlVarBinary) ||
+          is(column, mysqlCore.MySqlChar) ||
+          is(column, SQLiteText)) &&
+        typeof column.length === 'number'
```

Comment:
it's not any 'better', I just like the `Predicate` module, do as you wish
```ts
const _is = (that) => (col) => is(col, that)
const isTextColumn = Predicate.and(
  Predicate.some([
    _is(pgCore.PgChar),
    _is(pgCore.PgVarChar),
    // etc
  ]),
  Predicate.mapInput(Predicate.isNumber,(_: column) => _.length)
)

// later
if (isTextColumn(column)) {...}
```

Reply:
I prefer to keep it as similar to drizzle-zod as possible just in case I need to compare the code